### PR TITLE
Reduce Water Stun/Weaken Time

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Reduces the stun time and weaken time of TURF_WET_WATER by half.

## Why It's Good For The Game
Water is basically as good as slapping somebody with a baton in some cases, which logically makes absolutely zero sense. Should it knock you over and make you drop items? Yes. But should it knock you down for 6-8 seconds? No.

While this also nerfs the usage of water tanks by simplemobs (such as Terrors) there REALLY shouldnt be a reason that they can use a free 8x8 tile stun for 6-8 seconds.

edit: Slimes (the race) being able to leave stun-like landmines behind them when bleeding is also INSANELY silly from an Antag perspective.

## Changelog
:cl:
tweak: tweaked stun and weaken amounts in TURF_WET_WATER (Stun: 4 -> 2, Weaken: 2 -> 1)
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
You should include specifically in the changelog how much you tweaked/changed the values imo. Otherwise though, nice change.

**Comment by Dashsora** [comment]
This is FOR SURE a good change - but there's one thing i'd like to propose : What if all slips displace you one tile, and lube/acid naners three tiles, for the short stun duration, BUT if you hit an object like a wall/ closed locker/machine/table/etc, you get stunned for longer ?

It could check how many tiles you should've displaced, and how many tiles you actually displaced, and then handle the stun accordingly (More remaining tiles = longer stun, so if you slip on lube and immediately bonk your head on a wall you get stunned for longer than if it's on the end of the slide, water potential bonk stun would always be the same since it displaces only one tile)

**Comment by matttheficus** [comment]
> You should include specifically in the changelog how much you tweaked/changed the values imo. Otherwise though, nice change.

fixed

**Comment by Necaladun** [comment]
I'll miss being able to robust sec with just a bucket of water in perma but this is a needed change imo.

**Comment by MRgaty** [comment]
I think this will increment the amount of people that use space lube.

**Comment by matttheficus** [comment]
> I'll miss being able to robust sec with just a bucket of water in perma but this is a needed change imo.

you still CAN, it just wont stun for as long, so make sure to yoink their dropped weapons and stun them ASAP!

**Review by Kyep — APPROVED** [review]
This doesn't need my approval, as it already has a head voting for it.
Just for the record though, I approve too.

**Review by variableundefined — CHANGES_REQUESTED** [review]
Since this is an important variable that might be tweaked in the future, do you mind making a constant say, called 

#def WATER_STUN_TIME 
and 
#def WATER_WEAKEN TIME

And refer to that constant in the lines, and then #undef them at the end?

**Comment by matttheficus** [comment]
> Since this is an important variable that might be tweaked in the future, do you mind making a constant say, called
> 
> #def WATER_STUN_TIME
> and
> #def WATER_WEAKEN TIME
> 
> And refer to that constant in the lines, and then #undef them at the end?

Is it worth separating the values twice? The slip effect (at least the one changed here) is already only being applied to TURF_WET_WATER, implying that it only changes

**Comment by variableundefined** [comment]
@matttheficus 
Yes, magic number should be avoided. 

While this number is not reused anywhere, a meaningful name is going to be helpful for a maintainer, future contributor or you in the future to recall the **intent** behind the code. It also makes it easier for a layman to understand what is going on. 

If, for some reason I want to change the code for stun, I go into this file and I can see WATER_STUN_TIME in all caps at the top. I can search for the variable and quickly find the code responsible and debug it. 

Without it, I will have to search thoroughly for the code in this file that affects the water slip before I know what to do, and if a subtle typo is made, I will have a hard time understanding whether that is the intent in the first place.

**Comment by variableundefined** [comment]
@matttheficus 
Please enact requested changes within 4 days

**Comment by matttheficus** [comment]
life slammed stuff to a halt, changes made (hopefully properly, havent worked with defines at all with DM, used to constants instead)

**Review by variableundefined — APPROVED** [review]
@matttheficus You did it correctly

**Pull request merged**
